python/tk_framework_desktopserver/shotgun/scripts/execute_command.py

```diff
@@ -59,67 +52,87 @@ def core_info(engine):
-        logger.info(
-            "<b>You are up to date! There is no need to update the Toolkit "
-            "Core API at this time!</b>"
+        engine.log_info(
+            "*You are up to date! There is no need to update the Toolkit "
+            "Core API at this time!*"
         )
     elif status == TankCoreUpdater.UPDATE_BLOCKED_BY_SG:
         req_sg = installer.get_required_sg_version_for_update()
-        logger.warning(
-            "<b>A new version (%s) of the core API is available however "
-            "it requires a more recent version (%s) of Shotgun!</b>" % (lv, req_sg)
+        engine.log_warning(
+            "*A new version (%s) of the core API is available however "
+            "it requires a more recent version (%s) of Shotgun!*" % (lv, req_sg)
         )
     elif status == TankCoreUpdater.UPDATE_POSSIBLE:
         (summary, url) = installer.get_release_notes()
 
-        logger.info("<b>A new version of the Toolkit API (%s) is available!</b>" % lv)
-        logger.info("")
-        logger.info(
-            "<b>Change Summary:</b> %s <a href='%s' target=_new>"
-            "Click for detailed Release Notes</a>" % (summary, url)
+        engine.log_info("*A new version of the Toolkit API (%s) is available!*" % lv)
+        engine.log_info(
+            "*Change Summary:* %s [Click for detailed Release Notes](%s)" % (summary, url)
         )
-        logger.info("")
-        logger.info("In order to upgrade, execute the following command in a shell:")
-        logger.info("")
+        engine.log_info("In order to upgrade, execute the following command in a shell:")
 
         if sys.platform == "win32":
             tank_cmd = os.path.join(install_root, "tank.bat")
         else:
             tank_cmd = os.path.join(install_root, "tank")
 
-        logger.info("<code style='%s'>%s core</code>" % (code_css_block, tank_cmd))
-        logger.info("")
+        engine.log_info("*%s core*" % tank_cmd)
     else:
         raise sgtk.TankError("Unknown Upgrade state!")
 
+def pre_engine_start_callback(context):
+    """
+    The pre-engine-start callback that's given to the bootstrap API.
+    This callback handles attaching a custom logger to SGTK prior to
+    the Shotgun engine being initialized. This allows us to customize
+    the output of the logger in such a way that it is easily identified
+    and filtered before going back to the client for display.
+
+    :param context: The context object being used during the bootstrap
+        process.
+    """
+    import sgtk
+
+    sgtk.LogManager().initialize_base_file_handler("tk-shotgun")
+
+    # We need to make sure messages from this logger end up going to
+    # stdout. We'll be trapping stdout from the RPC API, which will
+    # give us the output that gets sent back to the client when the
+    # command is completed.
+    handler = logging.StreamHandler(sys.stdout)
+    sgtk.LogManager().initialize_custom_handler(handler)
```

Comment:
again, do we know we have the LogManager()? 

I was looking at this code last night (just before i handed over to you!), and i was thinking about a slightly different approach - curious what you think:

- in the `bootstrap()` method, call the `LogManager` get a good logging root for the sg engine. See [here](https://github.com/shotgunsoftware/tk-core/blob/master/python/tank/platform/engine.py#L114): `logger = LogManager.get_logger("env.%s.%s" % (env.name, engine_instance_name))` it's a bit hacky, but if we want to be backwards compatible it was the best thing i could think of. Can you think of anything better? That would give the `tk-shotgun` engine a standard logger writing to the correct place in the hierarchy. 

- Still in the `bootstrap()` metod - before we call the sgtk bootstrap, set up the log handler that prints to stdout. This is the place where we know exactly what core we are running so it's easier to write safe code.

- as we set up the params for bootstrappig, we make sure pass the logger we created above to the `pre_engine_start_callback` 

- `pre_engine_start_callback` just does one thing - hooks up the logger with `sgtk.log` - since we don't know what core runs in the `pre_engine_start_callback` we want to keep the footprint inside here as lightweight as possible.

This means logging starts earlier so we can track say errors at engine startup. It also has a minimal footprint in old-core territory.

what do you think?